components/helpers/applyFilter.ts

```diff
@@ -0,0 +1,156 @@
+  name: string;
+}
+
+interface Filter {
+  [key: string]: string;
+  value: string;
+}
+
+interface FilterOption {
+  value: string;
+  text: string;
+}
+
+/**
+ * @description Sorts an array of objects based on a string property called 'value'.
+ * @param arr Array of objects with a 'value' property of type string.
+ */
+export function sortFilter(arr: { value: string }[]): { value: string }[] {
+  return arr.sort((a, b) => {
+    if (a.value < b.value) {
+      return -1;
+    }
+    if (a.value > b.value) {
+      return 1;
+    }
+
+    return 0;
+  });
+}
+
+/**
+ * @description Apply filters to data and update the filters.
+ * @param checks Array of filter criteria objects.
+ * @param data Array of data objects to filter.
+ * @param setFilters Function to update the filters.
+ */
+export const applyFilterList = (checks: FilterCriteria[],
+  data: DataObject[],
+  setFilters: (lists: { [key: string]: FilterOption[] }) => void): void => {
+  if (Object.keys(checks).length) {
+    const lists: { [key: string]: FilterOption[] } = {};
+
+    checks.forEach((check) => {
+      lists[check.name] = [];
+    });
+    for (let i = 0; i < data.length; i++) {
+      const res = data[i];
+
+      Object.keys(lists).forEach((key) => {
+        const result = data[i][key];
+
+        if (res) {
+          if (lists[key].length) {
+            if (Array.isArray(result)) {
+              result.forEach((a) => {
+                if (a.name) {
+                  if (!lists[key].some((e) => e.value === a.name)) {
+                    const newData = {
+                      value: a.name,
+                      text: a.name
+                    };
+
+                    lists[key].push(newData);
+                    sortFilter(lists[key]);
+                  }
+                } else if (!lists[key].some((e) => e.value === a)) {
+                  const newData = {
+                    value: a,
+                    text: a
+                  };
+
+                  lists[key].push(newData);
+                  sortFilter(lists[key]);
+                }
+              });
+            } else if (!lists[key].some((e) => e.value === result)) {
+              const newData = {
+                value: result,
+                text: result
+              };
+
+              lists[key].push(newData);
+              sortFilter(lists[key]);
+            }
+          } else if (Array.isArray(result)) {
+            result.forEach((e) => {
+              if (e.name) {
+                const newData = {
+                  value: e.name,
+                  text: e.name
+                };
+
+                lists[key].push(newData);
+              } else {
+                const newData = {
+                  value: e,
+                  text: e
+                };
+
+                lists[key].push(newData);
+              }
+            });
+          } else {
+            const newData = {
+              value: result,
+              text: result
+            };
+
```

Comment:
Why there are blank lines in between the code? Kindly remove them from every file.

Reply:
Comment not addressed.

Reply:
i removed blank lines from many files but when i run lint:fix it gets added again in some files